File: 442. Find All Duplicates in an Array/442. Find All Duplicates in an Array.go
func findDuplicates(nums []int) []int {
    var rep []int
    sort.Ints(nums)
    for i := 0; i < len(nums) - 1; i++ {
        if nums[i] == nums[i + 1] {
            rep = append(rep, nums[i])
        }
    }
    return rep
}

File: 442. Find All Duplicates in an Array/442. Find All Duplicates in an Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> disnum;
        if (nums.empty()) {
            return disnum;
        }
        sort(nums.begin(), nums.end());
        for (int i = 0; i < nums.size() - 1; i++) {
            if (nums[i] == nums[i + 1]) {
                disnum.push_back(nums[i]);
            }
        }
        return disnum;
    }
};
